controllers/seedgen_controller.go

```diff
@@ -861,56 +921,65 @@ func (r *SeedGeneratorReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return
 	}
 
-	if isSeedGenFailed(seedgen) {
+	phase := getPhase(seedgen)
+
+	if phase == phaseFailed {
```

Comment:
probably can benefit from switch statement clarity and performance since we have 4 or more cases?

Reply:
Will do... That's what I wanted to do, but was looking up the wrong go function, and ended up just doing the if-checks :)